src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -145,52 +132,46 @@ public function syncPivot($model, $input)
      * Create any existing one to one relations for the current model from the form data.
      *
      * @param  \Illuminate\Database\Eloquent\Model  $item  The current CRUD model.
-     * @param  array  $data  The form data.
+     * @param  array  $input  The form data.
      */
-    private function createOneToOneRelations($item, $data)
+    private function createOneToOneRelations($item, $input)
     {
-        $relationData = $this->getRelationDataFromFormData($data);
-        $this->createRelationsForItem($item, $relationData);
+        $relationDetails = $this->getRelationDetailsFromInput($input);
+        $this->createRelationsForItem($item, $relationDetails);
     }
 
     /**
      * Create any existing one to one relations for the current model from the relation data.
      *
      * @param  \Illuminate\Database\Eloquent\Model  $item  The current CRUD model.
-     * @param  array  $formattedData  The form data.
+     * @param  array  $formattedRelations  The form data.
      * @return bool|null
      */
-    private function createRelationsForItem($item, $formattedData)
+    private function createRelationsForItem($item, $formattedRelations)
     {
-        if (! isset($formattedData['relations'])) {
+        if (! isset($formattedRelations['relations'])) {
             return false;
         }
-        foreach ($formattedData['relations'] as $relationMethod => $relationData) {
-            if (! isset($relationData['model'])) {
+        foreach ($formattedRelations['relations'] as $relationMethod => $relationDetails) {
+            if (! isset($relationDetails['model'])) {
                 continue;
             }
-            $model = $relationData['model'];
+            $model = $relationDetails['model'];
             $relation = $item->{$relationMethod}();
 
             if ($relation instanceof BelongsTo) {
-                $modelInstance = $model::find($relationData['values'])->first();
+                $modelInstance = $model::find($relationDetails['values'])->first();
                 if ($modelInstance != null) {
                     $relation->associate($modelInstance)->save();
                 } else {
                     $relation->dissociate()->save();
                 }
-            } elseif ($relation instanceof HasOne) {
-                if ($item->{$relationMethod} != null) {
-                    $item->{$relationMethod}->update($relationData['values']);
-                    $modelInstance = $item->{$relationMethod};
-                } else {
-                    $modelInstance = new $model($relationData['values']);
-                    $relation->save($modelInstance);
-                }
+            } elseif ($relation instanceof HasOne || $relation instanceof MorphOne) {
+                $modelInstance = $relation->updateOrCreate([], $relationDetails['values']);
```

Comment:
Great use of `updateOrCreate()` 👍  When I see 7 lines turning into 1 simple line... I know we've made progress 😀